uint8 algorithm
uint8 node_state
string node_state_desc
uint8 bug_state
string bug_state_desc
